android-client-sdk/src/test/java/com/devcycle/sdk/android/openfeature/DevCycleContextMapperTest.kt

```diff
@@ -29,6 +29,60 @@ class DevCycleContextMapperTest {
         assertEquals("user-123", jsonMap["userId"])
     }
 
+    @Test
+    fun `maps userId attribute to user ID when no targeting key`() {
+        val context = ImmutableContext(
+            attributes = mutableMapOf(
+                "userId" to Value.String("user-from-userId")
+            )
+        )
+        
+        val result = DevCycleContextMapper.evaluationContextToDevCycleUser(context)
+        
+        assertNotNull(result)
+        val jsonMap = convertToJsonMap(result!!)
+        assertEquals("user-from-userId", jsonMap["userId"])
+        assertEquals(false, jsonMap["isAnonymous"]) // User should be identified
+    }
+
+    @Test
+    fun `maps user_id attribute to user ID when no targeting key or userId`() {
+        val context = ImmutableContext(
+            attributes = mutableMapOf(
+                "user_id" to Value.String("user-from-user_id")
+            )
+        )
+        
+        val result = DevCycleContextMapper.evaluationContextToDevCycleUser(context)
+        
+        assertNotNull(result)
+        val jsonMap = convertToJsonMap(result!!)
+        assertEquals("user-from-user_id", jsonMap["userId"])
+        assertEquals(false, jsonMap["isAnonymous"]) // User should be identified
+    }
+
+    @Test
+    fun `prioritizes targeting key over userId and user_id attributes`() {
```

Comment:
Is this consistent with the other sdks? Afaik we use user_id as the priority on server side.